### PR TITLE
RDM-8251 - Separate display/entry regex; add exclusions to entry

### PR DESCRIPTION
**Pull request by danlysiak**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-8251

### Change description ###
Separate display/entry regex; add exclusions to entry

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeDisplayValidatorImpl.java [hunk 1]
```diff
@@ -13,6 +13,9 @@
 @Component
 public class DateTimeDisplayValidatorImpl implements DisplayContextParameterValidator {
 
+    Pattern NOT_ALLOWED_CHARACTERS_PATTERN_DATETIME = Pattern.compile("[VzOXxZ]");
```
Comment (kiran-yenigala-hmcts) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeDisplayValidatorImpl.java:16]: Please follow naming conventions, these variables are not static, final.... Then no need to define them in CAPITAL letters

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeEntryValidatorImpl.java [hunk 2]
```diff
@@ -13,6 +13,9 @@
 @Component
 public class DateTimeEntryValidatorImpl implements DisplayContextParameterValidator {
 
+    Pattern NOT_ALLOWED_CHARACTERS_PATTERN_DATETIME = Pattern.compile("[YDVzOXxZ]");
+    Pattern NOT_ALLOWED_CHARACTERS_PATTERN_DATE = Pattern.compile("[YDahKkHmsSAnNVzOXxZ]");
```
Comment (kiran-yenigala-hmcts) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeEntryValidatorImpl.java:17]: Please follow naming conventions, these variables are not static, final.... Then no need to define them in CAPITAL letters

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DisplayContextParameterValidator.java [hunk 3]
```diff
@@ -6,9 +6,6 @@
 
 public interface DisplayContextParameterValidator {
 
-    Pattern NOT_ALLOWED_CHARACTERS_PATTERN_DATETIME = Pattern.compile("[VzOXxZ]");
-    Pattern NOT_ALLOWED_CHARACTERS_PATTERN_DATE = Pattern.compile("[ahKkHmsSAnNVzOXxZ]");
-
     DisplayContextParameterType getType();
 
     void validate(String parameterValue, String fieldType) throws Exception;
```
Comment (kiran-yenigala-hmcts) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DisplayContextParameterValidator.java:11]: I just looked into this implementation. both implementations are doing the same thing..
    public void validate(final String parameterValue, String fieldType) throws InvalidDateTimeFormatException {
        Pattern pattern = ((fieldType.equals(BASE_DATE)) ? NOT_ALLOWED_CHARACTERS_PATTERN_DATE : NOT_ALLOWED_CHARACTERS_PATTERN_DATETIME);
        dateTimeFormatParser.parseDateTimeFormat(parameterValue, pattern);
    }

except the values of NOT_ALLOWED_CHARACTERS_PATTERN_DATE  and NOT_ALLOWED_CHARACTERS_PATTERN_DATETIME are changed.

we can provide some methods in the interface to get the values from the runtime object, like:  DisplayContextParameterType getType();

Pattern getDateTimePattern()
Pattern getDatePattern();

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeDisplayValidatorImpl.java (hunk 1 above)
Comment (danlysiak) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeDisplayValidatorImpl.java]: They can be static final now following the move. Adjusted, so no need to rename.

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeEntryValidatorImpl.java (hunk 2 above)
Comment (danlysiak) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeEntryValidatorImpl.java]: They can be static final now following the move. Adjusted, so no need to rename.

**Pull request merged**